### ISSUE TITLE
OpenSMTPD exits whenever I try to send any mail

### ISSUE THREAD

**Issue by DemiMarie** [OpenSMTPD/OpenSMTPD#1032]
After the recent patches, neither of my OpenBSD VMs can send any mail.  Removing the queue does not help, nor has anything else I have tried.

Any attempt at sending mail causes OpenSMTPD to exit immediately.

For what it is worth, neither VM has access to DNS, but that should not cause OpenSMTPD to crash, especially with no useful error message.

**Comment by DemiMarie** [comment]
Changing `/etc/resolv.conf` to disable DNS lookup via nameservers prevents this.  That said, exiting with absolutely no useful error message is still a bug.

**Comment by poolpOrg** [comment]
Indeed.

**Comment by ssartor** [comment]
Was this ever addressed?  I'm seeing the same issue using the latest portable release on NetBSD.

**Comment by DemiMarie** [comment]
@ssartor are you matching on `from socket`?  I was.

**Comment by ssartor** [comment]
@DemiMarie No, was just using the default conf file.  Using 'swaks' to generate a test email from the command line:

/usr/pkg/etc # swaks --to ssartor@odyssey.localnet --server localhost
=== Trying localhost:25...
=== Connected to localhost.
<-  220 odyssey.localnet ESMTP OpenSMTPD
 -> EHLO odyssey.localnet
<-  250-odyssey.localnet Hello odyssey.localnet [::1], pleased to meet you
<-  250-8BITMIME
<-  250-ENHANCEDSTATUSCODES
<-  250-SIZE 36700160
<-  250-DSN
<-  250 HELP
 -> MAIL FROM:<root@odyssey.localnet>
<-  250 2.0.0 Ok
 -> RCPT TO:<ssartor@odyssey.localnet>
<-  250 2.1.5 Destination address valid: Recipient ok
 -> DATA
<-  354 Enter mail, end with "." on a line by itself
 -> Date: Sun, 13 Sep 2020 11:28:22 -0400
 -> To: ssartor@odyssey.localnet
 -> From: root@odyssey.localnet
 -> Subject: test Sun, 13 Sep 2020 11:28:22 -0400
 -> Message-Id: <20200913112822.026401@odyssey.localnet>
 -> X-Mailer: swaks v20190914.0 jetmore.org/john/code/swaks/
 ->
 -> This is a test mailing
 ->
 ->
 -> .
*** Remote host closed connection unexpectedly.

This silently kills smtpd -- no error message, no core file.  This was on a NetBSD 9 system using the portable source code patched for the 'smtpctl encrypt' bug.
